### PR TITLE
feat(Banner): update Banner to use CSS Modules behind feature flag

### PR DESCRIPTION
**Pull request by joshblack**
Closes https://github.com/github/primer/issues/3882

#### Changed

- Refactor `Banner` to use CSS Modules behind a feature flag
- Update `Link` to no longer forward `underline` to the underlying node

### Rollout strategy

- [x] Minor release

### CODE REVIEW

File: packages/react/src/Banner/Banner.module.css [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    --banner-bgColor: var(--bgColor-attention-muted);
+    --banner-borderColor: var(--borderColor-attention-muted);
+    --banner-icon-fgColor: var(--fgColor-attention);
+  }
+}
+
+/* BannerContainer -------------------------------------------------------- */
+
+.BannerContainer {
+  font-size: var(--text-body-size-medium, 0.875rem);
```
Comment (jonrohan) [inline packages/react/src/Banner/Banner.module.css:48]: Could remove some of these fallbacks

File: packages/react/src/Banner/Banner.module.css [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  display: flex;
+  column-gap: var(--base-size-12, 0.5rem);
+  align-items: center;
+}
+
+.BannerActions :where([data-primary-action='trailing']) {
+  display: none;
+}
+
+@media screen and (min-width: 544px) {
```
Comment (jonrohan) [inline packages/react/src/Banner/Banner.module.css:136]:
```suggestion
@media screen and (--viewportRange-regular) {
```

This custom media query should be the same here, 🤞🏻

File: packages/react/src/Banner/Banner.module.css [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    display: flex;
+  }
+
+  .BannerActions :where([data-primary-action='leading']) {
+    display: none;
+  }
+}
+
+.Banner[data-dismissible] .BannerActions {
+  margin-block-end: var(--base-size-6, var(--base-size-6));
```
Comment (jonrohan) [inline packages/react/src/Banner/Banner.module.css:147]: weird fallback

File: packages/react/src/Banner/Banner.module.css [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  height: var(--base-size-16, 1rem);
+}
+
+/* BannerDismiss ---------------------------------------------------------- */
+
+.BannerDismiss {
+  display: grid;
+  place-items: center;
+  padding: var(--base-size-8, var(--base-size-8));
+  margin-inline-start: var(--base-size-4, var(--base-size-4));
```
Comment (jonrohan) [inline packages/react/src/Banner/Banner.module.css:117]: weird fallbacks

File: packages/react/src/Banner/Banner.module.css [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  font-size: inherit;
+  font-weight: var(--base-text-weight-semibold, 600);
+}
+
+/* BannerIcon ------------------------------------------------------------- */
+
+.BannerIcon {
+  display: grid;
+  place-items: center;
+  padding: var(--base-size-8, var(--base-size-8));
```
Comment (jonrohan) [inline packages/react/src/Banner/Banner.module.css:97]: weird fallbacks

**Review by jonrohan — APPROVED** [review]
Looks good ⚡

File: packages/react/src/Banner/Banner.module.css [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    --banner-icon-fgColor: var(--fgColor-attention);
+  }
+}
+
+/* BannerContainer -------------------------------------------------------- */
+
+.BannerContainer {
+  font-size: var(--text-body-size-medium);
+  align-items: start;
+  line-height: var(--text-body-lineHeight-medium));
```
Comment (langermank) [inline packages/react/src/Banner/Banner.module.css:50]:
```suggestion
  line-height: var(--text-body-lineHeight-medium);
```

**Pull request merged**